static/src/javascripts/projects/commercial/modules/messenger.js

```diff
@@ -18,32 +19,47 @@ define([
         unregister: unregister
     };
 
-    function register(type, callback, _window) {
+    function register(type, callback, options) {
+        options || (options = {});
+
         if( registeredListeners === 0 ) {
-            on(_window || window);
+            on(options.window || window);
         }
 
-        listeners[type] || (listeners[type] = []);
-        if (listeners[type].indexOf(callback) === -1) {
-            listeners[type].push(callback);
+        /* Persistent listeners are exclusive */
+        if (options.persist) {
+            listeners[type] = callback;
             registeredListeners += 1;
+        } else {
+            listeners[type] || (listeners[type] = []);
+            if (listeners[type].indexOf(callback) === -1) {
+                listeners[type].push(callback);
+                registeredListeners += 1;
+            }
         }
     }
 
-    function unregister(type, callback, _window) {
+    function unregister(type, callback, options) {
+        options || (options = {});
+
         if (callback === undefined) {
             registeredListeners -= listeners[type].length;
             listeners[type].length = 0;
         } else {
-            var idx = listeners[type].indexOf(callback);
-            if (idx > -1) {
+            if (listeners[type] === callback) {
+                listeners[type] = null;
                 registeredListeners -= 1;
-                listeners[type].splice(idx, 1);
+            } else {
+                var idx = listeners[type].indexOf(callback);
```

Comment:
I've guarded the whole process. If `listeners[type]` is defined, we know it is either a function or an array.